kotlin-inject-compiler/ksp/src/main/kotlin/me/tatarka/inject/compiler/ksp/InjectProcessor.kt

```diff
@@ -27,63 +28,69 @@ class InjectProcessor(
     private lateinit var provider: KSAstProvider
     private lateinit var injectGenerator: InjectGenerator
     private lateinit var kmpComponentCreateGenerator: KmpComponentCreateGenerator
-    private var deferredClasses: List<KSClassDeclaration> = mutableListOf()
-    private var deferredFunctions: List<KSFunctionDeclaration> = mutableListOf()
+    private var lastResolver: Resolver? = null
+    private var deferredClassNames: List<KSName> = mutableListOf()
+    private var deferredFunctionNames: List<KSName> = mutableListOf()
 
     private val kmpComponentCreateFunctionsByComponentType = mutableMapOf<AstClass, MutableList<AstFunction>>()
 
     override fun process(resolver: Resolver): List<KSAnnotated> {
+        lastResolver = resolver
         provider = KSAstProvider(resolver, logger)
         injectGenerator = InjectGenerator(provider, options)
         kmpComponentCreateGenerator = KmpComponentCreateGenerator(provider, options)
 
-        val previousDeferredClasses = deferredClasses
-        val previousDeferredFunctions = deferredFunctions
-
-        val componentSymbols = previousDeferredClasses + resolver.getSymbolsWithClassAnnotation(
```

Comment:
That's a behavior change. Shouldn't you still try to generate code for deferred classes from previous rounds and not wait until `finish()`? This would allow other symbol processor to react to the generated code.

Reply:
As per https://github.com/google/ksp/issues/1854#issuecomment-2161292017 `getSymbolsWithAnnotation` adds i the deferred symbols so we don't have to manually do it ourselves.

Reply:
Wouldn't this mean you generate the same component now twice? This would result in exceptions. E.g. round 1 symbol is deferred, round 2 you generate the code in `process()`, `finish()` gets called and you generate the class one more time.

Reply:
Oh never mind, it's 
```
deferredClassNames = deferredClasses.mapNotNull { it.qualifiedName }
```
And not
```
deferredClassNames += deferredClasses.mapNotNull { it.qualifiedName }
```